vmdk_plugin/drivers/shared/shared_driver.go

```diff
@@ -0,0 +1,209 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shared
+
+//
+// VMWare vSphere Shared Docker Data Volume plugin version.
+//
+// Provide support for --driver=shared in Docker, when Docker VM is running under ESX.
+//
+// Serves requests from Docker Engine related to vsphere shared volume operations.
+///
+
+import (
+	"fmt"
+	"path/filepath"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/docker/go-plugins-helpers/volume"
+	"github.com/vmware/docker-volume-vsphere/vmdk_plugin/utils/plugin_utils"
+	"github.com/vmware/docker-volume-vsphere/vmdk_plugin/utils/refcount"
+)
+
+const (
+	version = "vSphere Volume Shared Driver v0.1"
+)
+
+// VolumeDriver - vsphere shared volume driver struct
```

Comment:
We will have more shared-specific things here later.
It's the same as the VolumeDriver struct used for photon-specific and block-specific driver.